zookeeper/settings.sls

```diff
@@ -90,19 +115,15 @@
   {%- do zookeepers_with_ids.append(zookeeper_with_id)  %}
   {%- do connection_string.append( node.encode('ascii') + ':' + port | string() ) %}
   {%- do zookeepers.append( node.encode('ascii') ) %}
-  {%- if myid_dist|length == 0 %}    
-    {%- if node in (salt['network.get_hostname'](),
-                grains['id'],
-                grains['fqdn'],
-                grains['nodename']) 
-                or node in minion_ips %}
-      {%- do myid_dist.append(node_id)  %}
+  {%- if myid_tmp|length == 0 %}    
+    {%- if node in minion_ids %}
+      {%- do myid_tmp.append(node_id)  %}
```

Comment:
That's what I meant above, no need to mess around with `myid_tmp`. This is straight forward:
```jinja
  {%- if not myid_dist and node in minion_ids %}    
    {%- do myid_dist.append(node_id)  %}
  {%- endif %}
```